src/query/v2/plan/operator.cpp

```diff
@@ -2560,7 +2580,7 @@ class DistributedExpandCursor : public Cursor {
         auto &edge = *current_in_edge_it_;
         ++current_in_edge_it_;
         frame[self_.common_.edge_symbol] = edge;
-        pull_node(edge, EdgeAtom::Direction::IN);
+        PullDstVertex(frame, context);
```

Comment:
Why not to pass in the direction? Accessing the edge just to get an information that we already know seems unnecessary.

Reply:
We don't care about the direction. We care about fetching the properties of the dst vertex  (I know this is done for us on paths but we still have to fetch the last dst vertex). This is a slight hack, once we polish, we will use `GetProperties`

Reply:
My explanation is wrong, but I think the requested change is valid.

Inside the `PullDstVertex` function the `get_dst_vertex` is called with `self_.common_.direction`, but that can be `BOTH` not just `IN` or `OUT`. That's why the direction was passed in previously. The current implementation will fail if `self_.common_.direction` is `BOTH`.